### ISSUE TITLE
Invites part I

### ISSUE THREAD

**Issue by sudara** [sudara/alonetone#1088]
A starting place to allow moderators to create invite links. (#874 goes into all hopes and dreams...)

* Moderators can generate a unique invite link that can be used by many people and knows the number of active users that were created from it
* Moderators can generate a single use invite link which gets sent to particular email address. 
* When a prospective user opens a single use invite link, the email address is pre-filled on a user creation form. 
* Users created by an invite link should be attributed to the person who invited them (see: `invited_by_id` on User)
* Moderators can list active invites and it's clear which are single use vs. open ended
* Moderators can mark invites as inactive / invalid at which point they will no longer function
* Invalid invites redirect to `/get_an_account` with a flash message apology

```
invite
    email
    token
    invited_by_id
    single_use?
    users_created_count
    valid?
    created_at
```

Notes:

* Emails are unique to users on alonetone, so validation has to happen before invite is sent.

Questions:

* Email verification normally occurs after account creation before log in is allowed. Can it be skipped if the person is coming via a one-off emailed link?
* Is there a valid reason to store the `invite_id` on the user (or some other relationship)? We already store `invited_by_id` (because of account_requests and the plan to allow user to user invites).

**Comment by Manfred** [comment]
> Moderators can generate a unique invite link that can be used by many people and knows the number of active users that were created from it.

This is going to behave significantly different from a use-once `Invite` so I don't think we should merge them into one model. Can I call this a `GroupInvite`?

**Comment by sudara** [comment]
@Manfred Sure! I guess to some degree `GroupInvite` really is just a token that gives access to the "create new user" page that can be revoked and tracked. If we are going to separate the model, perhaps we can also add a `name` field so the front end can say "Welcome, RPM Challengers!" or "Welcome, r/wearethemusicmakers" etc....

**Comment by sudara** [comment]
Maybe `MassInvite` would be better, as a `Group` is another feature half-built, just lower priority than all the work that we have been doing.

**Comment by Manfred** [comment]
Yes, I like `MassInvite`.

**Comment by sudara** [comment]
Maybe this is out of scope for now, but ideally we'd be able to see the users who signed up via a mass invite link. `User.invited_by_id` is there to track moderator-approved accounts and future mod-to-user and user-to-user invites, but perhaps not "enough" for `MassInvites`

**Comment by sudara** [comment]
👏🏼 👏🏼 👏🏼
